### PR TITLE
Fixed globe icon placement in buttons by adding left padding

### PR DESCRIPTION
**Pull request by imajit**
Part of #9686 
In my last PR I had by reduced the button to the red part in the nav bar
Screenshot from 2021-06-15 16-55-32
I have corrected that part now the entire button is clickable in cases where globe icon is not there , here entire green part is now clickable 
Screenshot from 2021-06-15 16-56-09

For buttons I have added some left padding in the function,now the icons don't render inside the buttons and in cases where the button occupies the entire width globe icon is rendered below the button
**Admin Page Buttons** 
Screenshot from 2021-06-15 17-23-28

**Dashboard Buttons**
Screenshot from 2021-06-16 01-39-38

**Subscriptions Page Buttons**
Screenshot from 2021-06-16 01-48-07
The add button here is a `submit` button and has some tests linked with it. If you feel it is okay with the nested globe icon I can just leave it unchanged, else I'll change it with the globe icon outside the button and change the tests for it in a later PR.

### CODE REVIEW

**Comment by jywarren** [comment]
Hi @imajit this looks lovely! Thank you for your attention to detail as well!

I think we used to limit this to 5 max appearing on the screen so that it's not overwhelming. Did we ever implement that? I think this is fine but that's the only thing I worry a little about here. Do you perhaps want to address that soon too? I don't think it has to block merging this one. 

Thank you!!!

File: app/views/sidebar/_dashboard.html.erb
Comment (jywarren) [inline app/views/sidebar/_dashboard.html.erb:10]: Ooh, actually... this is interesting. This repetition does make for a lot more code... i wonder... what do you think, is there any way to reduce repetition in this while still achieving what we want? What are some ideas to try to brainstorm this? 

Sorry to pause a moment, but I hope my hesitation makes sense, @imajit -- still very much appreciate this work!

Comment (imajit) [reply app/views/sidebar/_dashboard.html.erb:10]: Yeah, I agree there is a repetition in the code as I'm wrapping the button inside a div so that the globe icon stays inline with the text. 
-  If I keep the `if` part only, then the button size reduces for all users
- If I keep the `else` part, then the globe icon overflows to the next line
- I can't store it in a new file as partial as the repetition  is kind of nested inside each `div`
So, I did this bit ugly implementation to satisfy all cases.  One way to avoid this is if I reduce the number of globe icons in the navbar, in that case, there will be less code repetition and fewer number of globe icons on the Dashboard. 
For now, I can't see a different way of implementation though :confused:

**Comment by imajit** [comment]
> Hi @imajit this looks lovely! Thank you for your attention to detail as well!
> 
> I think we used to limit this to 5 max appearing on the screen so that it's not overwhelming. Did we ever implement that? I think this is fine but that's the only thing I worry a little about here. Do you perhaps want to address that soon too? I don't think it has to block merging this one.
> 
> Thank you!!!

I did read about it in a discussion https://github.com/publiclab/plots2/issues/5651#issuecomment-706574770 but I don't think it has been implemented, there is a way using jquery to do this, 
`       jQuery(".translationClass").children(":gt(6)").hide();`
this will show only 5 globe icons, but I have never tested it on a live server with multiple users.

File: app/views/sidebar/_dashboard.html.erb
Comment (jywarren) [reply app/views/sidebar/_dashboard.html.erb:10]: Hmm.... I totally appreciate this is a tough one. Thanks for thinking hard on it. What if... we did it in JavaScript, after rendering? Could that work?

**Comment by jywarren** [comment]
> this will show only 5 globe icons, but I have never tested it on a live server with multiple users.

Did you want to try it in the JavaScript dev console on stable.publiclab.org to get a sense for it?

File: app/views/sidebar/_dashboard.html.erb
Comment (imajit) [reply app/views/sidebar/_dashboard.html.erb:10]: I added a helper function to render the dropdown button, this reduces some repetition

File: app/helpers/application_helper.rb
Comment (jywarren) [inline app/helpers/application_helper.rb:171]: Ah, this is much cleaner. Could we rename this though so it's a little more specific? Let me think --- 
```suggestion
  def create_nav_dropdown_item(href, text)
```

**Review by jywarren — CHANGES_REQUESTED** [review]
The helper function does look good. I suggested a slight renaming. Does this approach work for the lines in this partial too, like lines 11-26?

**Comment by jywarren** [comment]
Also, this will need conflicts resolve as well. Thanks for sticking with this @imajit -- much appreciated!

**Comment by imajit** [comment]
Hmm, I think this is due to some recently added tests , looking into

**Comment by jywarren** [comment]
Ah indeed. You can try resolving directly or rebasing!

File: app/helpers/application_helper.rb
Comment (jywarren) [reply app/helpers/application_helper.rb]: OK awesome. I think this is one of the last things before we merge, right? Thank you!!

Comment (imajit) [reply app/helpers/application_helper.rb]: Yeah

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Nice! Thanks for doing this so thoroughly, @imajit !!

**Pull request merged**
